---
float64 resolution
int32 width
int32 height
float64 origin_x
float64 origin_y